File: python/01/part1.py
def part1(x):
    with open(x, "r") as f:
        return max([sum([int(calorie) for calorie in calories.split('\n')])
                    for calories in f.read().split("\n\n")])


test_val = 24000
if part1("./test") == test_val:
    print('Solution is {}'.format(part1("./input")))
else:
    print("Test failed")

File: nim/01/part1.nim
import strutils

proc solution(path:string):int =
  let file = open(path)
  defer: file.close()

  var total = 0

  for line in file.lines():
    if line.len < 1:
        total = 0
        continue
    total += parseInt(line)
    if total > result: result = total

proc part1*(): void =
    let TEST = 24000
    if solution("test") == TEST:
        echo solution("input")